src/main/java/io/fabric8/maven/docker/StartMojo.java

```diff
@@ -483,6 +488,45 @@ private String getTcpHost(WaitConfiguration.TcpConfiguration tcpConfig, Properti
         return host;
     }
 
+    private WaitUtil.WaitChecker getHealthyWaitChecker(final String imageConfigDesc, final DockerAccess docker, final String containerId, final List<String> logOut) {
+
+
+        return new WaitUtil.WaitChecker() {
+
+            private boolean first = true;
+
+            @Override
+            public boolean check() {
+                try {
+                    final InspectedContainer container = docker.getContainer(containerId);
+                    if (container == null) {
+                        log.debug("HealthyWaitChecker:  Container %s not found");
+                        return false;
+                    }
+
+                    if (first) {
```

Comment:
Yes, I hope we can be sure since e.g. the [TcpWaitChecker](https://github.com/rhuss/docker-maven-plugin/blob/master/src/main/java/io/fabric8/maven/docker/StartMojo.java#L382-L383) also relies on it when it is created. Haven't got any bug report for this yet ;-) so it should work.